Utility/pmf.cpp

```diff
@@ -87,25 +85,32 @@ PMF::read_pmf(const std::string& myfile)
   }
   amrex::Print() << line_count << " data lines found in PMF file" << std::endl;
 
-  PMF::pmf_N = line_count;
-  PMF::pmf_M = variable_count - 1;
-  PMF::pmf_X.resize(PMF::pmf_N);
-  PMF::pmf_Y.resize(PMF::pmf_N * PMF::pmf_M);
+  pmf_data.pmf_N = line_count;
+  pmf_data.pmf_M = variable_count - 1;
+  pmf_data.pmf_X = (amrex::Real *) The_Arena()->alloc(pmf_data.pmf_N * sizeof(amrex::Real));
+  pmf_data.pmf_Y = (amrex::Real *) The_Arena()->alloc(pmf_data.pmf_N * pmf_data.pmf_M * sizeof(amrex::Real));
 
   iss.clear();
   iss.seekg(0, std::ios::beg);
   std::getline(iss, firstline);
   std::getline(iss, secondline);
-  for (int i = 0; i < PMF::pmf_N; i++) {
+  for (int i = 0; i < pmf_data.pmf_N; i++) {
     std::getline(iss, remaininglines);
     std::istringstream sinput(remaininglines);
-    sinput >> PMF::pmf_X[i];
-    for (int j = 0; j < PMF::pmf_M; j++) {
-      sinput >> PMF::pmf_Y[j * PMF::pmf_N + i];
+    sinput >> pmf_data.pmf_X[i];
+    for (int j = 0; j < pmf_data.pmf_M; j++) {
+      sinput >> pmf_data.pmf_Y[j * pmf_data.pmf_N + i];
     }
   }
 
-  PMF::d_pmf_X = PMF::pmf_X.dataPtr();
-  PMF::d_pmf_Y = PMF::pmf_Y.dataPtr();
+  pmf_data_g = (PmfData *) The_Device_Arena()->alloc(sizeof(pmf_data)); 
+#ifdef AMREX_USE_CUDA
```

Comment:
Do we need to add AMREX_USE_HIP conditions here as well?

Reply:
That is something I'm not entirely sure. Before we only had NVIDIA/CUDA device, now there are several options, but we still use the USE_CUDA = TRUE and AMREX_USE_CUDA everywhere. The hto2memcpy from AMReX is such that it works regardless, but maybe there is a more generic compiler `ifdef` we should use to see if we compile for GPU ?

Reply:
In the Tulip branch that has to be merged, I started the annoying task of doing 
#If defined(AMREX_USE_CUDA) || defined(AMREX_USE_HIP)
or
#If !defined(AMREX_USE_CUDA) && !defined(AMREX_USE_HIP)

We're going to have to propagate those generalize checks here too...

Reply:
Did you have to use a USE_HIP = TRUE in the makefile to have it compile with HIP ?

Reply:
Having a single compiler option to test if CUDA/HIP/DPC++ would be nice, rather than playing with logical combinations of three.

Reply:
Yes I did.  And yes, having something like AMREX_USE_GPU would be good. I'll suggest

Reply:
This already exists.

Reply:
Hey, at least I guessed correctly :)